blas/tpls/KokkosBlas1_rotmg_tpl_spec_decl.hpp

```diff
@@ -181,8 +181,24 @@ namespace Impl {
           KokkosBlas::Impl::CudaBlasSingleton::singleton();                    \
       KOKKOS_CUBLAS_SAFE_CALL_IMPL(                                            \
           cublasSetStream(s.handle, space.cuda_stream()));                     \
-      cublasDrotmg(s.handle, d1.data(), d2.data(), x1.data(), y1.data(),       \
-                   param.data());                                              \
+      typename DXView::HostMirror d1_h = Kokkos::create_mirror_view(d1);       \
+      typename DXView::HostMirror d2_h = Kokkos::create_mirror_view(d2);       \
+      typename DXView::HostMirror x1_h = Kokkos::create_mirror_view(x1);       \
+      typename YView::non_const_type::HostMirror y1_h =                        \
+          Kokkos::create_mirror_view(y1);                                      \
+      typename PView::HostMirror param_h = Kokkos::create_mirror_view(param);  \
+      Kokkos::deep_copy(d1_h, d1);                                             \
```

Comment:
If we're going to support exec space instances, might as well make all these deep_copies asynchronous.

Reply:
Also it might make sense to just do ``cublasSetPointerMode(s.handle, CUBLAS_POINTER_MODE_DEVICE)``. It's probably faster than many tiny host <-> device copies, although I haven't measured it.

Reply:
Wait, the transformations are applied to x1 and y1, right? Why are they being copied to host for cublas?

Reply:
I will do the `cublasSetPointerMode()` instead of copying everything to host and then back to device.
It should be better than what I am doing now and will avoid having to dance around the fact that y1 is actually `const`.

Reply:
Using `cublasSetPointerMode()` did the trick quite well!